### DISCUSSION TITLE
DataGridCheckBoxColumn not changing bound property until the item has lost focus

### DISCUSSION THREAD

**Discussion by PeterSchipperSogeti** [AvaloniaUI/Avalonia#12017] [Q&A] [answered]
**Describe the bug**
When you bind an object with a boolean property to a DataGridCheckBoxColumn, the property of the object does not change when expected. 
The update is only processed once the user selects another item in the DataGrid.

**To Reproduce**
Create a datagrid like this:

    <DataGrid Grid.Row="1" Name="MyDataGrid" Items="{Binding}">
    			<DataGrid.Columns>
    				<DataGridTextColumn Header="Forename"  Binding="{Binding Name}"/>
    				<DataGridTextColumn Header="ID" Binding="{Binding Id}" />
    				<DataGridCheckBoxColumn Binding="{Binding Bool}"></DataGridCheckBoxColumn>
    			</DataGrid.Columns>
    		</DataGrid>

And the Person object like this:

    public class Person
        {
            public string Name { get; set; }
            public int Id { get; set; }
    
            private bool a;
            public bool Bool {
                get
                {
                    return a;
                } 
                set 
                { 
                    a = value;
                } 
            }
        }

And the mainwindow like this:

    public partial class MainWindow : Window
        {
            public ObservableCollection<Person> Persons = new() {
                new(){
                    Name = "A",
                    Id = 1,
                    Bool = true,
                },
                new(){
                    Name = "B",
                    Id = 2,
                    Bool = false,
                }
            };
            public MainWindow()
            {
                InitializeComponent();
                MyDataGrid.DataContext = Persons;
                
            }
        }

Now set a breakpoint on the setter for Person.Bool.
When you start the app and check a box, you will notice the setter is not called. 
It is only called after you click another Person in the list.

**Expected behavior**
The setter of the Bool property of Person should be fired as soon as the checkbox is checked.

**Desktop (please complete the following information):**

- OS: Windows
- Version 19044.3086

**Comment by timunie** [comment, marked as answer]
This is how `DataGridCheckboxColumn` works. It only updates the binding when exit editing mode. Same is true for TextColumn. If you need direct impact, use a `TemplateColumn` instead. 

As this is the expected behavior for this control, I'm moving this issue into our Q&A section.

**Reply by ClassyCircuit** [reply]
Expected behavior or not, this is a very bad design and should be changed. Have you ever seen an application which required you to change focus to another property just to check a checkbox once more? I understand that `TemplateColumn` can be used as a workaround, but that means `DataGridCheckboxColumn` is broken by design.
